spec/jekyll/polyglot/patches/jekyll/site_spec.rb

```diff
@@ -163,6 +163,54 @@
     end
   end
 
+  describe @derive_lang_from_path do
+    before(:each) do
+      config = Jekyll.configuration(
+        'site' => @site,
+        'languages'                 => ['en', 'es', 'pt-br'], # fr not included
+        'default_lang'              => @default_lang,
+        'exclude_from_localization' => @exclude_from_localization,
+        'source'                    => File.expand_path('fixtures', __dir__),
+        'lang_from_path'            => true
+      )
+      @site = Site.new(config)
+      @site.prepare
+    end
+    it 'should derive lang from any part of the path' do
+      collection_en = Jekyll::Collection.new(@site, 'en')
+      collection_es = Jekyll::Collection.new(@site, 'es')
+      collection_pt_br = Jekyll::Collection.new(@site, 'pt-br')
+      collection_wrong = Jekyll::Collection.new(@site, 'wrong')
+      specs = {
+        'en' => [
+          Jekyll::Document.new('about.en.md', site: @site, collection: collection_en),
+          Jekyll::Document.new('fr/about.en.md', site: @site, collection: collection_en)
+        ],
+        'es' => [
```

Comment:
In spanish the name of the file should be `acercade`, or even `acerca_de`, since it is 2 words. Just to keep the name making sense.